File: polo_client/lib/src/polo_client_helper.dart
import 'dart:async';
import 'dart:convert';
import 'dart:io' as io;

part 'polo_client_base.dart';

/// Creates an Instance of Polo Websocket Client (`PoloClient`)
class Polo {
  /// Connects to the `PoloServer` and Returns the Instance of `PoloClient`
  static Future<PoloClient> connect(String url) async {
    io.WebSocket client = await io.WebSocket.connect(url);
    return PoloClient._(client);
  }
}

File: polo_client_ts/src/polo_client_helper.ts
import { PoloClient } from "./polo_client_base.ts";
import { Completer } from "./utils.ts";

class Polo {
  static async connect(url: string): Promise<PoloClient> {
    const ready: Completer<void> = new Completer<void>();

    const client: WebSocket = new WebSocket(url);
    client.onerror = (e) => {
      ready.reject((e as ErrorEvent).error);
    };
    client.onopen = (_) => {
      ready.complete();
    };

    await ready.promise;
    return PoloClient._(client);
  }
}

export { Polo };
